components/global/core/io.cellery.observability.api/src/main/java/io/cellery/observability/api/interceptor/AuthInterceptor.java

```diff
@@ -41,7 +42,10 @@ public boolean interceptRequest(Request request, Response response) {
         if (!request.getHttpMethod().equalsIgnoreCase(HttpMethod.OPTIONS) &&
                 request.getHeader(HttpHeaders.AUTHORIZATION) != null) {
             String header = request.getHeader(HttpHeaders.AUTHORIZATION);
-            String accessToken = header.split(" ")[1];
+            String cookieHeader = request.getHeader(HttpHeaders.COOKIE);
```

Comment:
There's a better way to get the cookie using MSF4J incoming request.
String cookieValue = request.getHeaders().getCookies().get(COOKIE_KEY);